data/BookingsDataExtracts/DataExtractWorker.cs

```diff
@@ -0,0 +1,36 @@
+using Nbs.MeshClient;
+using Microsoft.Extensions.Options;
+using Nbs.MeshClient.Auth;
+
+namespace BookingsDataExtracts;
+public class DataExtractWorker(
+    IHostApplicationLifetime hostApplicationLifetime,
+    IOptions<MeshSendOptions> meshSendOptions,
+    IOptions<MeshAuthorizationOptions> meshAuthOptions,
+    IMeshFactory meshFactory,
+    TimeProvider timeProvider,
+    BookingDataExtract bookingDataExtract
+    ) : BackgroundService
+{
+    protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+    {
+        var outputFile = new FileInfo(GenerateFileName());        
+        await bookingDataExtract.RunAsync(outputFile);
+
+        await SendViaMesh(outputFile);
+
+        hostApplicationLifetime.StopApplication();
+    }
+
+    private async Task SendViaMesh(FileInfo fileToSend)
+    {
+        if (string.IsNullOrEmpty(meshSendOptions.Value.DestinationMailboxId) == false)
```

Comment:
In the spirit of failing fast or not failing silently, should this blow up if MailboxId is null?